cfme/base/__init__.py

```diff
@@ -121,7 +123,14 @@ def get_master(self):
             server = server_collection.all[0]
         else:
             server = server_collection.find_by(is_master=True)[0]
-        return self.instantiate(name=server.name, sid=server.id)
+
+        try:
+            name = server.name
+        except AttributeError:
+            logger.error('The EVM has no name, setting it to EVM')
+            name = 'EVM' if self.appliance.version == LATEST else server.name
```

Comment:
**Optional**

Use `not self.appliance.is_downstream` instead of importing `LATEST` and checking version against it.

**Question**
If server.name just produced an AttributeError, if you're just expecting it to reraise, consider breaking up the ternary and setting name if upstream, else explicitly reraise.